helix-core/src/main/java/org/apache/helix/controller/rebalancer/DelayedAutoRebalancer.java

```diff
@@ -408,4 +420,129 @@ private int getNumExtraReplicas(ClusterConfig clusterConfig) {
     }
     return numExtraReplicas;
   }
+
+
+  private Map<String, String> computeBestPossibleStateForPartition(ResourceControllerDataProvider cache,
```

Comment:
This function has a lot of redundant code with existing logic. I would suggest to simplify it.

Reply:
As I wrote in my change description, I will make the dedup effort as next change. Trying to make code easy to read and reason through. There are 3 more changes planned (but this is the core-logic). Other 3 changes are: 1. dedup code. 2. move the process() logic to DataProcessing stage  and (3) add the metrics. 

So can you please review for "CORRECTNESS"

Reply:
As we talked offline, I have done the code cleanup/consolidation with this change. the line of code change have increased but between the 2 versions, you can review core logic + code consolidation. thanks,